azure/datalake/store/lib.py

```diff
@@ -294,10 +143,18 @@ def __init__(self, store_name=default_store, token=None,
             __version__)
         self.req_timeout_s = req_timeout_s
 
+    def get_refreshed_bearer_token(self):
+        # Check if the token is about to expire in 300 seconds and refresh it if necessary
+        if self.AccessToken is None or time.time() > self.AccessToken.expires_on - 300:
+            self.AccessToken = self.token_credential.get_token(self.scopes)
+        return self.AccessToken.token
```

Comment:
FWIW, when using MSAL or its upper-level library Azure Identity, you can simply call their token acquisition method (`get_token()` in this case) unconditionally, and the library will take care of the refresh logic. I believe that is the same for Azure Identity. CC: @xiangyan99 

```suggestion
        self.AccessToken = self.token_credential.get_token(self.scopes)
        return self.AccessToken.token
```

Reply:
My local test code using DefaultAzureCredential was eventually ending up on using AzurePowershellCredential which doesn't support 
In-memory token caching according to the documentation. It seems to depend on the kind of token. 

See table 
https://github.com/Azure/azure-sdk-for-python/blob/ded844291d88a9fb2d4877f332fcc84e7361b6d3/sdk/identity/azure-identity/TOKEN_CACHING.md#credentials-supporting-token-caching

Reply:
OK, then. I'll defer to @xiangyan99 :-)

Reply:
Discussed with Xiang and we do want some caching here in case customer is using one of the methods without caching. and there is no harm in adding caching for other methods.